pkg/asset/machines/master.go

```diff
@@ -648,9 +665,61 @@ func (m *Master) Generate(ctx context.Context, dependencies asset.Parents) error
 			Data:     data,
 		}
 	}
+
+	// This is only used by Two Node OpenShift with Fencing (TNF)
+	// The credentials are rendered into secrets that will be consumed by the
+	// cluster-etcd operator to enable recovery via fencing
+	if pool.Fencing != nil && pool.Fencing.Credentials != nil {
+		credentials, err := gatherFencingCredentials(pool.Fencing.Credentials)
+		if err != nil {
+			return err
+		}
+
+		secrets, err := createSecretAssetFiles(credentials, fencingCredentialsSecretFileName)
+		if err != nil {
+			return fmt.Errorf("failed to gather fencing credentials for control plane hosts: %w", err)
+		}
+		m.FencingCredentialsSecretFiles = append(m.FencingCredentialsSecretFiles, secrets...)
+	}
+
 	return nil
 }
 
+func gatherFencingCredentials(credentials []*types.Credential) ([]corev1.Secret, error) {
+	secrets := []corev1.Secret{}
+
+	for _, credential := range credentials {
+		if credential.HostName == "" || credential.Username == "" || credential.Password == "" || credential.Address == "" {
```

Comment:
```suggestion
		if credential == nil || credential.HostName == "" || credential.Username == "" || credential.Password == "" || credential.Address == "" {
```

Just in case?